pkg/googlegroups/client.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlegroups
+
+import (
+	"context"
+	"fmt"
+
+	admin "google.golang.org/api/admin/directory/v1"
+	"google.golang.org/api/cloudidentity/v1"
+
+	"github.com/abcxyz/team-link/pkg/groupsync"
+)
+
+// TODO: take token to authenticate.
+// The token should be generated by WIF (google-github-actions/auth)
+// Currently we don't have a github action set up, we will use default
+// login credentials during dev phase.
+func NewGoogleGroupsReader(ctx context.Context) (groupsync.GroupReader, error) {
```

Comment:
nit: since this in the package `googlegroups` it would read a bit nice if this function was just called `NewGroupReader`. That way calling it outside the package would look like `googlegroups.NewGroupReader`

Reply:
Yes, I have considered this, the reason for naming it like this is there's a ` NewGroupReader(identityService *cloudidentity.Service, adminService *admin.Service)` exists already.

I changed this function name to `googlegroups.NewReader` so it makes its caller function `client.NewReader`.